### DISCUSSION TITLE
BitmapPalette equivalent in ImageSharp

### DISCUSSION THREAD

**Discussion by Veriara** [SixLabors/ImageSharp#1365] [Q&A] [answered]
I'm porting a `.NET Framework 4.6.2` application to `.NET Core 3.1` and it has proven to be quite a challenge when it comes to graphics, as the old application relies heavily on `System.Windows.Media` and the `PresentationCore` methods.

ImageSharp has done a great job at solving most of these issues, however I'm struggling to find a way to functionally replace `System.Windows.Media.Imaging.BitmapPalette`. Does anyone have any idea how I can achieve this functionality with ImageSharp?

Thank you for taking the time to read this. :)

**Comment by JimBobSquarePants** [comment, marked as answer]
Normally we would recommend users utilize the color cutting algorithms in our Wu and Octree quantizers to create the palette to use when saving images in indexed pixels formats.

However we do have a [`PaletteQuantizer`](https://docs.sixlabors.com/api/ImageSharp/SixLabors.ImageSharp.Processing.Processors.Quantization.PaletteQuantizer.html) class with which you can assign a custom palette then pass that quantizer instance to IImageEncoder instances allowing encoding in indexed format (_currently BMP, PNG, GIF_).
